src/trace/layers/ordered.rs

```diff
@@ -38,36 +39,37 @@ impl<K: Ord+Clone, L: Trie> Trie for OrderedLayer<K, L> {
 			let child_lower = self.offs[lower];
 			let child_upper = self.offs[lower + 1];
 			OrderedCursor {
-				keys: self.keys.clone(),
-				offs: self.offs.clone(),
+				keys: owned_self.clone().map(|x| &x.keys),
+				offs: owned_self.clone().map(|x| &x.offs),
```

Comment:
That makes sense! After posting the questions, I hopped on to #rust and asked if there was a type that takes an arbitrary `T: Deref` into another `S: Deref<Target=<T as Deref>::Target::...>`, and (i) got no answer, but (ii) realized that this would probably require a closure in the resulting type signature, which puts it off limits for now.